### ISSUE TITLE
[Feature Request] Override Geofence name as address if device is within a geofence

### ISSUE THREAD

**Issue by himsashu** [traccar/traccar#4615]
I want geofence name to be shown in the address field if the device is within that geofence. The reverse geocoded address is not to be used in this case. Is there any possibility in the traccar. I have been using this feature in opengts where I can enable or disable the reverse geocoding when I am creating or editing a geofence. The geofence table has a separate field for this purpose in opengts.
This feature might be useful in many cases. For example, the address is showing "home/office" if vehicle is within the geofence of my home/office. Currently address column is showing complete address from reverse geocoding.

**Comment by tananaev** [comment]
You can already see current geofence for each device. Seems like the value of this feature is very limited.

You mentioned "useful in many cases". Can you please elaborate. What cases it covers that current functionality doesn't?

**Comment by himsashu** [comment]
Yes you are right that I can see the geofence of each device but issue is with the different reports (trip report or stop report).
It might be more useful to show the start and end address as geofence name.
For example if my vehicle is started from my home and the trip is ended at my office. So in current scenario the start address and end address in trip report or stop report show the complete postal address (fetched from reverse geocoding). It might make more sense in this cases that if the start address show as HOME (geofence name) and end address show as OFFICE (geofence name).

**Comment by tananaev** [comment]
Makes sense. What if there are multiple geofences? How do you think it should be configurable?

**Comment by himsashu** [comment]
In case of multiple geofence, either both geofence name should be displayed in address or last/inner geofence address to be considered. The last geofence means the geofence in which the last arrival exist.

**Comment by sboldt1978** [comment]
+1! I would also _really_ like to see this feature becoming implemented. This would be of great help to track my trips and see how often I take certain trips troughout the year. If I can help implementing this feature, I'd love to. Can you give me a hint where in the code the start and stop locations are retrieved for trip report generation and how to check inside which geofences a location is located? If multiple geofences apply I would suggest to use the smallest in size, because it might be hard to find out which geofence was entered last.

**Comment by neuvill** [comment]
Dear team, is this feature been implemented ?

**Comment by tananaev** [comment]
No, as you can probably see from the status of this ticket.
